projects/plugins/jetpack/extensions/blocks/podcast-player/components/podcast-player.js

```diff
@@ -211,6 +216,18 @@ export class PodcastPlayer extends Component {
 		this.props.unregisterMediaSource( this.props.playerId );
 	}
 
+	componentDidUpdate( prevProps ) {
+		const trackGuids = tracks => ( tracks?.length ? tracks.map( track => track.guid ) : [] );
+		const guids = trackGuids( this.props.tracks );
+		const prevGuids = new Set( trackGuids( prevProps.tracks ) );
+
+		// This equality check is a bit rough. It relies on the guids being unique for example, but
+		// it should be fine for our requirements.
+		if ( guids.length !== prevGuids.size || ! guids.every( guid => prevGuids.has( guid ) ) ) {
+			this.registerPlayer( 0 );
```

Comment:
Although the action does almost the same, it shouldn't register a new media source but update the current one.